include/superior_mysqlpp/dnsa_connection_pool.hpp

```diff
@@ -101,88 +125,99 @@ namespace SuperiorMySqlpp
             DnsAwarePoolManagement& operator=(DnsAwarePoolManagement&&) = delete;
 
         private:
+            /**
+             * Returns reference to base class where is this mixin class used.
+             */
             Base& getBase()
             {
                 return *static_cast<Base*>(this);
             }
 
+            /**
+             * Returns const reference to class where is this mixin class used.
+             */
             const Base& getBase() const
             {
                 return *static_cast<const Base*>(this);
             }
 
-            /*
+            /**
+             * This function is called when unexpected error is thrown during dns resolution error handling.
+             * It's called only if logger throws exception.
+             */
+            void onUnexpectedError(void) const noexcept {
+                if(terminateOnFailure) {
+                    std::terminate();
+                }
+            }
+
+            /**
              * This function is run as parallel thread.
              * We cannot afford to let any exceptions to be propagated from logging functions
-             * since it would result in call to std::terminate()!!! Hence all the try/catch blocks.
+             * since it would result in call to std::terminate().
+             */
+            void job(void) const {
```

Comment:
Explicit void in parameters is completely redundant in C++ (not in C)

Reply:
ACK